File: benchmarks/python/binary_search.py
#!/usr/bin/env python3

# Binary Search benchmark in Python

def binary_search(arr, left, right, x):
    while left <= right:
        mid = left + (right - left) // 2
        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            left = mid + 1
        else:
            right = mid - 1
    return -1

# Create sorted array
size = 100000
arr = list(range(size))

# Perform multiple binary searches
count = 0
for i in range(0, size, 100):
    result = binary_search(arr, 0, size - 1, i)
    if result != -1:
        count += 1

print(f"Found {count} elements")

File: benchmarks/javascript/binary_search.js
// Binary Search benchmark in JavaScript

function binarySearch(arr, left, right, x) {
    while (left <= right) {
        const mid = left + Math.floor((right - left) / 2);
        if (arr[mid] === x) {
            return mid;
        } else if (arr[mid] < x) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1;
}

// Create sorted array
const size = 100000;
const arr = new Array(size);
for (let i = 0; i < size; i++) {
    arr[i] = i;
}

// Perform multiple binary searches
let count = 0;
for (let i = 0; i < size; i += 100) {
    const result = binarySearch(arr, 0, size - 1, i);
    if (result !== -1) {
        count++;
    }
}

console.log(`Found ${count} elements`);
